scipy/stats/tests/test_mstats_basic.py

```diff
@@ -201,6 +201,14 @@ def test_spearmanr(self):
               0.0, 0.6, 6.7, 3.8, 1.0, 1.2, 1.4, np.nan]
         (x, y) = (ma.fix_invalid(x), ma.fix_invalid(y))
         assert_almost_equal(mstats.spearmanr(x,y)[0], 0.6887299)
+        # Next test is to make sure calculation uses sufficient precision.
+        # The denominator's value is ~n^3 and used to be represented as an
+        # int. 2000**3 > 2**32 so these arrays would cause overflow on
+        # some machines.
+        x = np.arange(2000, dtype=np.float)
+        y = np.arange(2000, dtype=np.float)
+        y = np.concatenate((y[1000:], y[:1000]))
+        assert_almost_equal(mstats.spearmanr(x,y)[0], -0.50000037)
```

Comment:
I think I've got a way to make the theoretically correct value exactly 0.998 (i.e. 1 - 1/500). How does that sound?